### PR TITLE
ci: store version

### PR DESCRIPTION
**Pull request by DouglasBlackwood**
Change-Id: I5a8b82c4efeafde3bc9a92f17ae1376ce20091e9

### CODE REVIEW

**Comment by DouglasBlackwood** [comment]
This pull request is part of a stack:
1. ci: store version ([#310](https://github.com/Mergifyio/mergify-cli/pull/310))
1. feat: display version ([#311](https://github.com/Mergifyio/mergify-cli/pull/311))

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -18,10 +18,18 @@ jobs:
       - name: Get version
         run: |
           echo "version=$(date +'%Y.%m.%d.%H.%M')" >> "$GITHUB_ENV"
-      - run: |
+      - name: Bump version and push to GitHub
+        run: |
+          pip install poetry
+          poetry version $version
+          git config --global user.email "
```
Comment (Greesb) [inline .github/workflows/release.yml:25]: missing an email ?

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "mergify_cli"
-version = "0.0.0"
+version = "2023.09.27.08.14"
```
Comment (jd) [inline pyproject.toml:3]: is there a meaning?

**Review by jd — CHANGES_REQUESTED** [review]
not a fan of this approach, it's always error prone. Isn't there something based on git tags?

File: pyproject.toml (hunk 2 above)
Comment (DouglasBlackwood) [reply pyproject.toml:3]: It's the current version number, which is the date of release.

**Comment by DouglasBlackwood** [comment]
> not a fan of this approach, it's always error prone. Isn't there something based on git tags?

The goal is to be able to display the version number (see #311 ). I don't see how git tags could help.

File: .github/workflows/release.yml (hunk 1 above)
Comment (DouglasBlackwood) [reply .github/workflows/release.yml:25]: Yes, but is it useful?

**Comment by DouglasBlackwood** [comment]
> not a fan of this approach, it's always error prone. Isn't there something based on git tags?

The release system is based on git tags. Here, I want to bump the `pyproject.toml` file also to be able to display it using `mergify --version` (see #311 ).

File: pyproject.toml (hunk 2 above)
Comment (jd) [reply pyproject.toml:3]: Is it date + time? That's the first time I see this :|

**Comment by jd** [comment]
Yeah my point would be to avoid to have to bump the version manually with a  tool like [setuptools_scm](https://pypi.org/project/setuptools-scm/)

File: pyproject.toml (hunk 2 above)
Comment (DouglasBlackwood) [reply pyproject.toml:3]: Yeah, it's the date and the time (hours and minutes).
It was introduced here #249, specified by [this task](https://linear.app/mergify/issue/MRGFY-2664/fix-mergify-cli-pypi-release). We probably discussed the version system orally or on Slack. I don't find any track of it.

**Comment by DouglasBlackwood** [comment]
> Yeah my point would be to avoid to have to bump the version manually with a tool like [setuptools_scm](https://pypi.org/project/setuptools-scm/)

OK, I didn't get it at first, but the version is correctly set below with `poetry self add poetry-version-plugin` (https://github.com/tiangolo/poetry-version-plugin). We just can't retrieve it locally as HEAD is not the latest tag.

**Pull request closed without merging**